src/query/src/sql/show.rs

```diff
@@ -74,7 +75,11 @@ fn create_sql_options(table_meta: &TableMeta) -> Vec<SqlOption> {
         options.push(sql_option("compaction_time_window", number_value(w)));
     }
 
-    for (k, v) in &table_opts.extra_options {
+    for (k, v) in table_opts
+        .extra_options
+        .iter()
+        .filter(|(k, _)| k != &IMMUTABLE_TABLE_META_KEY)
+    {
         options.push(sql_option(k, string_value(v)));
```

Comment:
The order in which `FORMAT` and `LOCATION` are retrieved from the iterator is not deterministic. So, I sorted `self.options` in the `src/sql/src/statements/create.rs`